idea/src/org/jetbrains/kotlin/idea/inspections/collections/UselessCallOnCollectionInspection.kt

```diff
@@ -77,7 +78,7 @@ class UselessCallOnCollectionInspection : AbstractUselessCallInspection() {
                 KotlinBundle.message("useless.call.on.collection.type"),
                 ProblemHighlightType.LIKE_UNUSED_SYMBOL,
                 isOnTheFly,
-                RemoveUselessCallFix()
+                if (KotlinBuiltIns.isArray(receiverType)) ReplaceUselessCallWithToListFix() else RemoveUselessCallFix()
```

Comment:
That's wrong logic. You should use `ReplaceUselessCallWithToListFix` in case when type of whole expression is `List` and when receiver type isn't `List` at the same time. E.g. consider such example:
```
val arr: List<Int> = setOf(1, 2, 3).filterNotNull()
```
Your quick fix will remove `filterNotNull` but it's expected to be replaced with `toList` just like with `arrayOf`